src/mappers/mapAny.ts

```diff
@@ -50,9 +62,37 @@ export default function mapAny(context: ParseContext, node: Base): Node {
     return mapUndefined(context, node);
   }
 
+  if (node instanceof Block) {
+    return mapBlock(context, node);
+  }
+
+  if (node instanceof Code) {
+    return mapCode(context, node);
+  }
+
+  if (node instanceof While && !(node instanceof For)) {
```

Comment:
In `CoffeeScript`, `For` nodes extend `While` for a reason I don't fully understand.